File: matlab/functions/compute_std_obj.m
%% computes the std of the objective history

function [std_obj] = compute_std_obj(obj_list,history_size)
std_obj = Inf;
if numel(obj_list)>=history_size
    std_obj = std(obj_list(end-history_size+1:end));
    
    % normalize 
    std_obj = std_obj/abs(obj_list(end));
end
return;

File: python/functions/compute_std_obj.py
# Compute standard deviation
import numpy as np

def compute_std_obj(obj_list, history_size):
    std_obj = np.inf
    if len(obj_list) >= history_size:
        std_obj = np.std(obj_list[(len(obj_list)-history_size):])
        
        std_obj = std_obj/np.abs(obj_list[-1])
    return std_obj